src/main/java/frc/robot/utils/ThreadedGyro.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.utils;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ThreadedGyro {
+    private final AHRS gyro;
+    private final AtomicBoolean shouldReset = new AtomicBoolean(false);
+    private volatile double lastGyro;
```

Comment:
Every variable that is shared between threads should be `Atomic*`

Reply:
I think volatile would have worked because the each thread is either reading or writing (we dont have any operations that both read and write at same time like i++) but it does not hurt. I am now using an atomicLong to represent the byte value of the double because there is no such thing as an atomic double/float. Also, I did not want to use an atomic reference because if for some reason we want to compare to values and use the methods in atomic reference it would compare equality not identity (I think).